chain/chain/src/chain.rs

```diff
@@ -2891,98 +2900,187 @@ impl<'a> ChainUpdate<'a> {
+                            ChunkExtra::new(
+                                &apply_result.new_root,
+                                outcome_root,
+                                apply_result.validator_proposals,
+                                apply_result.total_gas_burnt,
+                                gas_limit,
+                                apply_result.total_balance_burnt,
+                            ),
+                        );
+                        chain_update.chain_store_update.save_outgoing_receipt(
+                            &block_hash,
+                            shard_id,
+                            apply_result.receipt_result,
+                        );
+                        // Save receipt and transaction results.
+                        chain_update.chain_store_update.save_outcomes_with_proofs(
+                            &block_hash,
+                            shard_id,
+                            apply_result.outcomes,
+                            outcome_paths,
+                        );
+                    }
+                    Err(err) => {
+                        let msg = if let Some(msg) = err.downcast_ref::<&'static str>() {
+                            msg.to_string()
+                        } else if let Some(msg) = err.downcast_ref::<String>() {
+                            msg.clone()
+                        } else {
+                            format!("?{:?}", err)
+                        };
+                        panic!("{}", msg);
+                    }
+                }
+            }
+
+            for result in dif_height_handlers {
+                let res = result.join();
+                match res {
+                    Ok(res) => {
+                        let (shard_id, apply_result) =
+                            res.map_err(|e| ErrorKind::Other(e.to_string()))?;
+
+                        let mut new_extra = chain_update
+                            .chain_store_update
+                            .get_chunk_extra(&prev_block_hash, shard_id)?
+                            .clone();
+
+                        chain_update
+                            .chain_store_update
+                            .save_trie_changes(apply_result.trie_changes);
+                        *new_extra.state_root_mut() = apply_result.new_root;
+
+                        chain_update.chain_store_update.save_chunk_extra(
+                            &block_hash,
+                            shard_id,
+                            new_extra,
+                        );
+                    }
+                    Err(err) => {
+                        let msg = if let Some(msg) = err.downcast_ref::<&'static str>() {
+                            msg.to_string()
+                        } else if let Some(msg) = err.downcast_ref::<String>() {
+                            msg.clone()
+                        } else {
+                            format!("?{:?}", err)
+                        };
```

Comment:
is this part necessary? Could we print the error directly?

Reply:
You mean leaving just the "else" part? We are not really printing here, we are propagating any panic that happened inside